common/clientCache.go

```diff
@@ -57,6 +58,14 @@ func NewClientCache(
 	}
 }
 
+func (c *clientCacheImpl) GetHostInfoForClient(key string) (string, error) {
```

Comment:
Should this be called GetHostNameForKey instead ? The input is key and the output is hostname